Description: Link with the GNUstep libraries on !Darwin.
Author: Yavor Doganov <yavor@gnu.org>
Last-Update: 2014-07-04

--- a/plugins/alarm_speech/uwsgiplugin.py
+++ b/plugins/alarm_speech/uwsgiplugin.py
@@ -10,6 +10,6 @@
     LIBS = ['-framework appkit']
 else:
     CFLAGS = ['-I /usr/include/GNUstep']
-    LIBS = []
+    LIBS = ['-lgnustep-gui -lgnustep-base -lobjc']
 
 GCC_LIST = ['alarm_speech.m']
