Description: Define directory where packaged plugins are installed.
Author: Leonid Borisenko <leo.borisenko@gmail.com>

Index: uwsgi-0.9.6/uwsgiconfig.py
===================================================================
--- uwsgi-0.9.6.orig/uwsgiconfig.py	2010-08-31 20:57:06.000000000 +0300
+++ uwsgi-0.9.6/uwsgiconfig.py	2010-08-31 20:59:53.000000000 +0300
@@ -31,7 +31,7 @@
 DEBUG=False
 EMBED_PLUGINS=True
 UWSGI_BIN_NAME = 'uwsgi'
-UWSGI_PLUGIN_DIR = '.'
+UWSGI_PLUGIN_DIR = '/usr/lib/uwsgi/plugins'
 
 
 # specific compilation flags
@@ -429,7 +429,7 @@
 	p_cflags.insert(0, '-I.')
 
 	plugin_base = path + '/' + up.NAME + '_plugin'
-	plugin_dest = UWSGI_PLUGIN_DIR + '/' + up.NAME + '_plugin'
+	plugin_dest = './' + up.NAME + '_plugin'
 
 	shared_flag = '-shared'
 
