Description: Enable integrate profiler.
Author: Leonid Borisenko <leo.borisenko@gmail.com>
Last-Update: 2010-08-30

Index: uwsgi-0.9.6/uwsgiconfig.py
===================================================================
--- uwsgi-0.9.6.orig/uwsgiconfig.py	2010-08-30 02:25:08.000000000 +0300
+++ uwsgi-0.9.6/uwsgiconfig.py	2010-08-30 02:25:14.000000000 +0300
@@ -11,7 +11,7 @@
 MULTICAST=True
 THREADING=True
 SENDFILE=True
-PROFILER=False
+PROFILER=True
 NAGIOS=True
 PROXY=True
 PASTE=True
