Description: Enable routing.
Author: Leonid Borisenko <leo.borisenko@gmail.com>
Last-Update: 2010-08-30

Index: uwsgi-0.9.6/uwsgiconfig.py
===================================================================
--- uwsgi-0.9.6.orig/uwsgiconfig.py	2010-08-30 02:14:01.000000000 +0300
+++ uwsgi-0.9.6/uwsgiconfig.py	2010-08-30 02:14:14.000000000 +0300
@@ -21,8 +21,8 @@
 HTTP=True
 EVDIS=False
 LDAP=True
-WSGI2=False
-ROUTING=False
+WSGI2=True
+ROUTING=True
 STACKLESS=False
 #PLUGINS = ['psgi']
 PLUGINS = []
