Description: Enable SCTP.
Author: Leonid Borisenko <leo.borisenko@gmail.com>
Last-Update: 2010-09-01

Index: uwsgi-0.9.6/uwsgiconfig.py
===================================================================
--- uwsgi-0.9.6.orig/uwsgiconfig.py	2010-09-01 11:59:58.000000000 +0300
+++ uwsgi-0.9.6/uwsgiconfig.py	2010-09-01 12:00:05.000000000 +0300
@@ -3,7 +3,7 @@
 XML=True
 INI=True
 SNMP=True
-SCTP=False
+SCTP=True
 ERLANG=True
 SPOOLER=True
 EMBEDDED=True
