Description: Fix interpreter path in 'ugreenchat.py'.
Author: Leonid Borisenko <leo.borisenko@gmail.com>
Last-Update: 2011-01-17

Index: uwsgi/ugreenchat.py
===================================================================
--- uwsgi-0.9.6.6.orig/ugreenchat.py	2011-01-17 07:29:02.000000000 +0200
+++ uwsgi-0.9.6.6/ugreenchat.py	2011-01-17 07:29:11.000000000 +0200
@@ -1,4 +1,4 @@
-#!./uwsgi
+#!/usr/bin/uwsgi
 import uwsgi
 
 from bottle import route, default_app, request, response, redirect
