Index: uwsgi/plugins/lua/uwsgiplugin.py
===================================================================
--- uwsgi.orig/plugins/lua/uwsgiplugin.py	2011-04-29 04:16:51.000000000 +0300
+++ uwsgi/plugins/lua/uwsgiplugin.py	2011-04-29 04:17:00.000000000 +0300
@@ -3,7 +3,7 @@
 try:
         LUALIB = os.environ['UWSGICONFIG_LUALIB']
 except:
-        LUALIB = 'lua'
+        LUALIB = 'lua5.1'
 
 try:
         LUAINC = os.environ['UWSGICONFIG_LUAINC']
