Description: Don't link plugin with unnecessary core libraries
Origin: upstream, http://projects.unbit.it/hg/uwsgi/rev/e61c6756d5e7
Last-Update: 2011-05-11

Index: uwsgi/uwsgiconfig.py
===================================================================
--- uwsgi.orig/uwsgiconfig.py	2011-05-11 17:30:28.000000000 +0300
+++ uwsgi/uwsgiconfig.py	2011-05-11 17:30:56.000000000 +0300
@@ -478,11 +478,10 @@
 
     p_cflags = cflags[:]
     p_ldflags = ldflags[:]
-    p_libs = libs[:]
 
     p_cflags += up.CFLAGS
     p_ldflags += up.LDFLAGS
-    p_libs += up.LIBS
+    p_libs = up.LIBS
 
     p_cflags.insert(0, '-I.')
 
