From: "roberto@quantal64" <roberto@quantal64>
Date: Wed Jul 25 19:47:15 2012 +0200
Subject: allows ugreen with threads
Origin: http://projects.unbit.it/hg/uwsgi-1.2/rev/8ee4797038aa


diff -r a6150bd4e1b8 -r 8ee4797038aa plugins/python/python_plugin.c
--- a/plugins/python/python_plugin.c	Wed Jul 25 19:46:13 2012 +0200
+++ b/plugins/python/python_plugin.c	Wed Jul 25 19:47:15 2012 +0200
@@ -1405,7 +1405,9 @@
 #ifndef UWSGI_PYPY
 void uwsgi_python_suspend(struct wsgi_request *wsgi_req) {
 
+	PyGILState_STATE pgst = PyGILState_Ensure();
 	PyThreadState *tstate = PyThreadState_GET();
+	PyGILState_Release(pgst);
 
 	if (wsgi_req) {
 		up.current_recursion_depth[wsgi_req->async_id] = tstate->recursion_depth;
@@ -1615,7 +1617,9 @@
 #ifndef UWSGI_PYPY
 void uwsgi_python_resume(struct wsgi_request *wsgi_req) {
 
+	PyGILState_STATE pgst = PyGILState_Ensure(); 
 	PyThreadState *tstate = PyThreadState_GET();
+	PyGILState_Release(pgst);
 
 	if (wsgi_req) {
 		tstate->recursion_depth = up.current_recursion_depth[wsgi_req->async_id];

