From: "roberto@quantal64" <roberto@quantal64>
Date: Wed Jul 25 19:46:13 2012 +0200
Subject: fixed env clear in plain async mode
Origin: http://projects.unbit.it/hg/uwsgi-1.2/rev/a6150bd4e1b8


diff -r 518b4ba97f83 -r a6150bd4e1b8 plugins/python/wsgi_handlers.c
--- a/plugins/python/wsgi_handlers.c	Wed Jul 25 19:45:18 2012 +0200
+++ b/plugins/python/wsgi_handlers.c	Wed Jul 25 19:46:13 2012 +0200
@@ -341,6 +341,7 @@
 
 #ifdef UWSGI_ASYNC
 	if (wsgi_req->async_status == UWSGI_AGAIN) {
+		wi = &uwsgi_apps[wsgi_req->app_id];
 		UWSGI_GET_GIL
 		// get rid of timeout
 		if (wsgi_req->async_timed_out) {
@@ -359,6 +360,7 @@
 			PyDict_SetItemString(wsgi_req->async_environ, "uwsgi.ready_fd", Py_None);
 		}
 		int ret = manage_python_response(wsgi_req);
+		if (ret == UWSGI_OK) goto end; 
 		UWSGI_RELEASE_GIL
 		return ret;
 	}
@@ -508,6 +510,7 @@
 	}
 
 	// this object must be freed/cleared always
+end:
 	if (wsgi_req->async_input) {
                 Py_DECREF((PyObject *)wsgi_req->async_input);
         }
