From: "roberto@quantal64" <roberto@quantal64>
Date: Sat Jul  7 10:37:21 2012 +0200
Subject: fixed logfile inheritance
Origin: http://projects.unbit.it/hg/uwsgi-1.2/rev/f6f09fc80883

diff -r eb63ba0814d8 -r f6f09fc80883 uwsgi.c
--- a/uwsgi.c	Sat Jul 07 10:35:46 2012 +0200
+++ b/uwsgi.c	Sat Jul 07 10:37:21 2012 +0200
@@ -1602,7 +1602,9 @@
 		}
         }
 	else if (uwsgi.logfile) {
-		logto(uwsgi.logfile);
+		if (!uwsgi.is_a_reload || uwsgi.log_reopen) {
+			logto(uwsgi.logfile);
+		}
 	}
 
 	if (uwsgi.never_swap) {
@@ -1839,7 +1841,9 @@
 	}
 
 	if (uwsgi.logto2) {
-		logto(uwsgi.logto2);
+		if (!uwsgi.is_a_reload || uwsgi.log_reopen) {
+			logto(uwsgi.logto2);
+		}
 	}
 
 	if (uwsgi.chdir) {
