From: "roberto@quantal64" <roberto@quantal64>
Date: Fri Jul 13 12:57:18 2012 +0200
Subject: fixed casting in wsgi.input
Origin: http://projects.unbit.it/hg/uwsgi-1.2/rev/e7bca57267bc

diff -r 85ae8fc31920 -r e7bca57267bc plugins/python/wsgi_subhandler.c
--- a/plugins/python/wsgi_subhandler.c	Fri Jul 13 12:56:03 2012 +0200
+++ b/plugins/python/wsgi_subhandler.c	Fri Jul 13 12:57:18 2012 +0200
@@ -51,7 +51,7 @@
         // if async_post is mapped as a file, directly use it as wsgi.input
         if (wsgi_req->async_post) {
 #ifdef PYTHREE
-                wsgi_req->async_input = PyFile_FromFd(fileno(wsgi_req->async_post), "wsgi_input", "rb", 0, NULL, NULL, NULL, 0);
+                wsgi_req->async_input = PyFile_FromFd(fileno((FILE *)wsgi_req->async_post), "wsgi_input", "rb", 0, NULL, NULL, NULL, 0);
 #else
                 wsgi_req->async_input = PyFile_FromFile(wsgi_req->async_post, "wsgi_input", "r", NULL);
 #endif
