From: Roberto De Ioris <roberto@unbit.it>
Date: Sat, 17 Mar 2012 10:36:06 +0100
Subject: python 3.x fix

---
 plugins/python/pyutils.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/plugins/python/pyutils.c b/plugins/python/pyutils.c
index 40a7bb4..c935b7e 100644
--- a/plugins/python/pyutils.c
+++ b/plugins/python/pyutils.c
@@ -71,7 +71,7 @@ void init_pyargv() {
 	}
 
 #ifdef PYTHREE
-	wchar_t *pname = uwsgi_calloc(strlen(argv0)+1);
+	wchar_t *pname = uwsgi_calloc(sizeof(wchar_t) * (strlen(argv0)+1));
 	mbstowcs(pname, argv0, strlen(argv0)+1);
 #else
 	char *pname = argv0;
-- 
1.7.9.1

