Description: Remove PHP libraries when building PHP plugin.
Author: Janos Guljas <janos@resenje.org>
Last-Update: 2012-06-16

--- a/plugins/php/uwsgiplugin.py
+++ b/plugins/php/uwsgiplugin.py
@@ -24,6 +24,6 @@
     LDFLAGS.append('-L%s' % ld_run_path)
     os.environ['LD_RUN_PATH'] = ld_run_path
 
-LIBS = [os.popen(PHPPATH + ' --libs').read().rstrip(), '-lphp5']
+LIBS = ['-lphp5']
 
 GCC_LIST = ['php_plugin']
