From: "roberto@quantal64" <roberto@quantal64>
Date: Thu May 31 18:51:29 2012 +0200
Subject: added a security check in the http router
Origin: http://projects.unbit.it/hg/uwsgi-1.2/rev/f204f6d12fac

diff -r 3ef3a10e8404 -r f204f6d12fac plugins/http/http.c
--- a/plugins/http/http.c	Thu May 31 13:38:07 2012 +0200
+++ b/plugins/http/http.c	Thu May 31 18:51:29 2012 +0200
@@ -492,6 +492,12 @@
 		hv = hv->next;
 	}
 
+	// security check
+	if (c >= MAX_HTTP_VEC-4) {
+		uwsgi_log("too much headers in request. skipping it.\n");
+		return 0;
+	}
+
 #ifdef UWSGI_DEBUG
 	uwsgi_log("vec size: %d pkt size: %d load %d\n", c, h_session->uh.pktsize, uhttp.load);
 #endif
